docs/adr/0004-instrumentation.md

```diff
@@ -0,0 +1,100 @@
+# Implement instrumentation via a new API Endpoint with external ad hoc probe calculation
+
+- Status: proposed
+- Deciders: Christina Harlow, Linh Nguyen, Rob Miller
+- Date: 2022-04-05
+
+Technical Story: https://github.com/mozilla/glam/issues/1847
+
+## Context and Problem Statement
+
+In order to help us with various product decisions and overall product quality,
```

Comment:
I would enumerate the problem statements (goals) here, then the requirements (in decision drivers) and then provide proposed solutions in the considered options section. E.g.

1.  better understanding glam usage (i.e. probe usage over time)
2. reducing the runtime of glam ETL
3. reduce costs (bq)?

Instead of conflating the solutions to these goals in the same section